tests/repository_simulator.py

```diff
@@ -192,24 +195,32 @@ def fetch(self, url: str) -> Iterator[bytes]:
         """Fetches data from the given url and returns an Iterator (or yields
         bytes).
         """
-        if not self.root.consistent_snapshot:
-            raise NotImplementedError("non-consistent snapshot not supported")
         path = parse.urlparse(url).path
         if path.startswith("/metadata/") and path.endswith(".json"):
+            # figure out rolename and version
             ver_and_name = path[len("/metadata/") :][: -len(".json")]
-            # only consistent_snapshot supported ATM: timestamp is special case
-            if ver_and_name == "timestamp":
-                version = None
-                role = "timestamp"
-            else:
+            # root is always version-prefixed while timestamp is always NOT
+            if ver_and_name.endswith("root") or (
```

Comment:
This will fail if a delegated role name ends with "root". But otherwise it is hard to generalize and I guess we don't want to support that weird cases in RepositorySimulator?

Reply:
I tried to do it in a readable way ... at least now it is correct for the cases I can think of.